### PR TITLE
WIP Created student page to resolve 404 error.

### PR DESCRIPTION
**Pull request by AlysonDick**
404 error received after clicking on "View details" on student table when logged in as admin. Created student page to resolve.

Added student preference data table to student page which includes the student preferences.

Changes also include new role and stage based access models for project and updates to preference buttons.

### CODE REVIEW

File: src/server/routers/user/student/preference.ts
Comment (AlysonDick) [inline src/server/routers/user/student/preference.ts:1]: Renaming data to studentProjectDetails (line 32) may be more clear, happy to discuss.

**Review by AlysonDick — COMMENTED** [review]
Couple of comments to discuss. otherwise happy to merge.

File: src/app/(protected)/[group]/[subGroup]/[instance]/projects/_components/projects-columns.tsx
[the remark this answers is not in the document]
Comment (pkitazos) [reply src/app/(protected)/[group]/[subGroup]/[instance]/projects/_components/projects-columns.tsx:1]: Yes, I'll remove them in the next commit

File: src/app/(protected)/[group]/[subGroup]/[instance]/students/[id]/_components/student-preference-action-menu.tsx
[the remark this answers is not in the document]
Comment (pkitazos) [reply src/app/(protected)/[group]/[subGroup]/[instance]/students/[id]/_components/student-preference-action-menu.tsx:1]: Yes, we should. Could you push up that fix?

**Pull request merged**